### PR TITLE
front: use Date in TrainScheduleWithDetails

### PR DESCRIPTION
**Pull request by emersion**
Instead of passing around formatted strings, pass a Date object. That way we don't need to undo the formatting in TimetableTrainCard later on, and we get improved type safety.

No behavior change intended.

### CODE REVIEW

File: front/src/modules/trainschedule/components/Timetable/TimetableTrainCard.tsx [hunk 1]
```diff
@@ -34,6 +34,9 @@ type TimetableTrainCardProps = {
   projectionPathIsUsed: boolean;
 };
 
+const formatFullDate = (d: Date) => dayjs(d).format('D/MM/YYYY HH:mm:ss');
+const formatDateHours = (d: Date) => dayjs(d).format('HH:mm:ss');
```
Comment (emersion) [inline front/src/modules/trainschedule/components/Timetable/TimetableTrainCard.tsx]: It would probably be better to use the `Intl` locale instead of dayjs for formatting here, because that would then use the locale's preferred date style instead hardcoding the French format for everybody. However that'd be a change of behavior so I'd prefer to keep it in a separate patch.

**Review by Akctarus — APPROVED** [review]
tested, LGTM, nice refactor :)

**Review by theocrsb — APPROVED** [review]
LGTM and tested!

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:
